README.md

```diff
@@ -185,9 +215,261 @@ docker run 348099934012.dkr.ecr.eu-central-1.amazonaws.com/kilt/prototype-chain
+- revoke a whole hierarchy `revoke_root`
+- lookup a root `get(root)`
+- lookup a delegation `get(delegation)`
+- lookup children of a delegation `get(children)`
+on chain.
+
+#### Create root
+```rust
+create_root(origin, root_id: T::DelegationNodeId, ctype_hash: T::Hash) -> Result
+```
+The `create_root` function takes the following parameters:
+
+- origin: The caller of the method, i.e., public address (ss58) of the owner of the
+ trust hierarchy
+- root_id: A V4 UUID identifying the trust hierarchy
+- ctype_hash: The blake2b hash of the CTYPE the trust hierarchy is associated with
+
+The node verifies the transaction signature and insert it to the state. The root is stored by using
+a map:
+
+```rust
+T::DelegationNodeId => (T::Hash,T::AccountId,bool)
+```
+
+#### Add delegation
+```rust
+add_delegation(origin, delegation_id: T::DelegationNodeId, root_id: T::DelegationNodeId, parent_id: Option<T::DelegationNodeId>, delegate: T::AccountId, permissions: Permissions, delegate_signature: T::Signature) -> Result
+```
+
+The `add_delegation` function takes the following parameters:
+- origin: The caller of the method, i.e., public address (ss58) of the delegator
+- delegation_id: A V4 UUID identifying this delegation
+- root_id: A V4 UUID identifying the associated trust hierarchy
+- parent_id: Optional, a V4 UUID identifying the parent delegation this delegation is
+ based on
+- CTYPEHash: The blake2b hash of CTYPE used when creating the Claim
+- delegate: The public address (ss58) of the delegate (ID receiving the delegation)
+- permissions: The permission bit set (having 0001 for attesting permission and
+ 0010 for delegation permission)
+- delegate_signature: ed25519 based signature by the delegate based on the
+ delegationId, rootId, parentId and permissions
+
+The node verifies the transaction signature and the delegate signature as well as all other data
+to be valid and the delegator to be permitted and then inserts it to the state. The delegation is
+stored by using a map:
+```rust
+T::DelegationNodeId => (T::DelegationNodeId,Option<T::DelegationNodeId>,T::AccountId,Permissions,bool)
+```
+
+Additionally, if the delegation has a parent delegation, the information about the children of its
+parent is updated in the following map that relates parents to their children:
+
+```rust
+T::DelegationNodeId => Vec<T::DelegationNodeId>
+```
+
+#### Revoke
+```rust
+revoke_root(origin, root_id: T::DelegationNodeId) -> Result
+```
+and
+```rust
+revoke_delegation(origin, delegation_id: T::DelegationNodeId) -> Result
+```
+
+## Updating to latest substrate-node-template
 
-The command `substrate-node-new`, described in https://docs.substrate.dev/docs/creating-a-custom-substrate-chain downloads a node-template, which this repo bases on.
+The command `substrate-node-new` included in the substrate installation, described in https://substrate.dev/docs/en/tutorials/creating-your-first-substrate-chain downloads a node-template, which this repo bases on.
```

Comment:
Not your text I know, but this sentence reads weird.
How about: 

The command `substrate-node-new` (included in the substrate installation and described in https://substrate.dev/docs/en/tutorials/creating-your-first-substrate-chain) downloads a node-template which this repo is based on.

Reply:
Gosh obvious one

Reply:
👍 please also replace --> "is based on"